1.5 enhancer

arschmitz opened the pull request:
A new stand alone auto enhancment module for declarative widgets.

arschmitz committed:
Enhancer: Add initial module

arschmitz committed:
Enhancer: Fixup tests add ns remove degrde inputs, links, keepNative

Also removes a bunch of deprecated code

arschmitz committed:
Enhance: Breakup

arschmitz committed:
Popup: fixup addition of hook

arschmitz committed:
Enhancer: add backcompat

arschmitz commented on js/widgets/enhancer.backcompat.js:
```diff
@@ -0,0 +1,52 @@
+	$.fn.enhance.initGenerator = generator;
+
+}
+});
```
missing new line

arschmitz commented on js/widgets/enhancer.js:
```diff
@@ -0,0 +1,110 @@
+});
+
+return plugin;
+} );
```
missing new line

gabrielschulhof commented on js/widgets/enhancer.backcompat.js:
```diff
@@ -0,0 +1,52 @@
+
+			if ( $.mobile.ignoreContentEnabled ) {
+				elements.each( function() {
+					if ( $( this ).closest( "[data-" + $.mobile.ns + "enhanced='false']" ).length ) {
```
I think the name of the attribute is supposed to be ```data-enhance```, not ```data-enhanced```

gabrielschulhof replied:
https://github.com/jquery/jquery-mobile/blob/master/js/helpers.js#L142

gabrielschulhof commented on js/widgets/widget.backcompat.js:
```diff
@@ -42,6 +42,7 @@ if ( $.mobileBackcompat !== false ) {
 		_create: function() {
 			this._setInitialOptions();
 			this._super();
+
```
Is it worth touching this file for only one blank line?

arschmitz replied:
there were other changes this is the results of a rebase that already incorporated them ( apparently minus this one line change )

arschmitz committed:
Enhancer: fixup breakup

arschmitz committed:
Page: change option enhance to enhanceWithin

We cant use the name enhance because it potentially conflicts with
data-enhance=false for ignoreContentEnabled

arschmitz committed:
DegradeInputs: Remove code supporting an IE8 bug that is no longer

reproducible with current code.

arschmitz committed:
Enhancer: Add tests for enhancer

arschmitz committed:
Enhancer: fixup new lines

arschmitz committed:
Enhancer: fixup style issues

arschmitz committed:
degradeInputs: Style fixes

arschmitz committed:
Build: add more files to jscs:good

arschmitz commented on tests/integration/forms/index.html:
```diff
@@ -84,7 +84,7 @@
 				<div data-nstest-role="fieldcontain">
 					<label for="testRange">Test Range</label>
-					<input id="testRange" type="range" min="0" max="100" value="42"></input>
+					<input id="testRange" type="number" data-nstest-role="slider" min="0" max="100" value="42"></input>
```
revert this

arschmitz commented on js/widgets/widget.backcompat.js:
```diff
@@ -44,6 +44,7 @@ if ( $.mobileBackcompat !== false ) {
 				this._super();
 				if ( !this.options.enhanced && this.options.wrapperClass ) {
 					this._addClass( this.widget(), null, this.options.wrapperClass );
+
```
revert this

arschmitz commented on tests/unit/slider/index.html:
```diff
@@ -37,32 +37,32 @@
 	<div id="foo" data-nstest-role="page">
 		<div data-nstest-role="fieldcontain">
-			<input type="range" name="slider-1" id="range-slider-up" value="0" min="0" max="100" data-nstest-theme="b" data-nstest-track-theme="a" />
+			<input type="number" data-nstest-role="slider" name="slider-1" id="range-slider-up" value="0" min="0" max="100" data-nstest-theme="b" data-nstest-track-theme="a" />
```
revert all these

arschmitz commented:
Still some usages of `.trigger( "create" )` in demos that need to be cleaned up

arschmitz commented:
We had talked about removing our usage of custom `initSelectors` as part of this. This PR does not do that though. Maybe i should add that and move all the custom ones to backCompat before we land this. That does involve a lot of work for the demos and tests though...

gabrielschulhof commented on js/widgets/enhancer.backcompat.js:
```diff
@@ -0,0 +1,53 @@
+/*!
+ * jQuery Mobile Enhnacer @VERSION
```
Enhancer

gabrielschulhof replied:
"Enhancer Backcompat" maybe so it's not the same as the actual enhancer.

gabrielschulhof commented on js/widgets/enhancer.js:
```diff
@@ -0,0 +1,115 @@
+/*!
+ * jQuery Mobile Enhnacer @VERSION
```
Enhancer

gabrielschulhof commented on tests/unit/degrade-inputs/degradeInputs.js:
```diff
@@ -3,9 +3,79 @@
 
-test( 'keepNative elements should not be degraded', function() {
-	deepEqual( $( 'input#not-to-be-degraded' ).attr( "type" ), "range" );
+test( "degrdeInputs works on page init", function() {
```
degrade

gabrielschulhof commented on tests/unit/degrade-inputs/degradeInputs.js:
```diff
@@ -3,9 +3,79 @@
+		"setting an input type to false cause no degradation of that type" );
+} );
+
+test( "degrdeInputs works on page enhance", function() {
```
degrade

arschmitz commented:
landed in 1.5-dev
